homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,179 @@
+"""Cover Platform for Dio Chacon REV-SHUTTER devices."""
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api.const import DeviceTypeEnum, ShutterMoveEnum
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import DioChaconDataUpdateCoordinator
+from .entity import DioChaconEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        DioChaconCover(coordinator, device)
+        for device in coordinator.list_devices
+        if device["type"] == DeviceTypeEnum.SHUTTER.value
+    )
+
+
+class DioChaconCover(DioChaconEntity, CoverEntity):
+    """Object for controlling a Dio Chacon cover."""
+
+    _attr_should_poll = False
+    _attr_assumed_state = True
+    _attr_has_entity_name = False
+
+    def __init__(
+        self,
+        coordinator: DioChaconDataUpdateCoordinator,
+        device,
+    ) -> None:
+        """Initialize the cover."""
+
+        target_id = device["id"]
+        name = device["name"]
+        openlevel = device["openlevel"]
+        movement = device["movement"]
+        connected = device["connected"]
+        model = device["model"]
+        device_class = CoverDeviceClass.SHUTTER
+
+        _LOGGER.info(
+            "Adding DIO Chacon SHUTTER Cover with id=%s, name=%s, openlevel=%s, movement=%s and connected=%s",
+            target_id,
+            name,
+            openlevel,
+            movement,
+            connected,
+        )
+
+        super().__init__(coordinator, target_id, name, model, connected)
```

Comment:
This can be just a lot lot shorter by not assigning to a variable before use. Also, don't log on info.

Reply:
Done